extensions/standard-processors/processors/ListenSyslog.cpp

```diff
@@ -17,318 +14,272 @@
-  _serverTheadRunning = true;
-  _thread = new std::thread(run, this);
-  _thread->detach();
-}
+  context->getProperty(MaxBatchSize.getName(), max_batch_size_);
+  context->getProperty(ParseMessages.getName(), parse_messages_);
 
-void ListenSyslog::run(ListenSyslog *process) {
-  process->runThread();
-}
+  uint64_t max_queue_size = 0;
+  context->getProperty(MaxQueueSize.getName(), max_queue_size);
+  max_queue_size_ = max_queue_size > 0 ? std::optional<uint64_t>(max_queue_size) : std::nullopt;
 
-void ListenSyslog::runThread() {
-  while (_serverTheadRunning) {
-    if (_resetServerSocket) {
-      _resetServerSocket = false;
-      // need to reset the socket
-      std::vector<int>::iterator it;
-      for (it = _clientSockets.begin(); it != _clientSockets.end(); ++it) {
-        int clientSocket = *it;
-        close(clientSocket);
-      }
-      _clientSockets.clear();
-      if (_serverSocket > 0) {
-        close(_serverSocket);
-        _serverSocket = 0;
-      }
-    }
+  Protocol protocol = Protocol(ProtocolProperty.getDefaultValue());
+  if (!context->getProperty(ProtocolProperty.getName(), protocol)) {
+    logger_->log_error("Missing or invalid Protocol: defaulting to %s", protocol.toString());
+  }
 
-    if (_serverSocket <= 0) {
-      uint16_t portno = _port;
-      struct sockaddr_in serv_addr;
-      int sockfd;
-      if (_protocol == "TCP")
-        sockfd = socket(AF_INET, SOCK_STREAM, 0);
-      else
-        sockfd = socket(AF_INET, SOCK_DGRAM, 0);
-      if (sockfd < 0) {
-        logger_->log_error("ListenSysLog Server socket creation failed");
-        break;
-      }
-      bzero(reinterpret_cast<char *>(&serv_addr), sizeof(serv_addr));
-      serv_addr.sin_family = AF_INET;
-      serv_addr.sin_addr.s_addr = INADDR_ANY;
-      serv_addr.sin_port = htons(portno);
-      if (bind(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
-        logger_->log_error("ListenSysLog Server socket bind failed");
-        break;
-      }
-      if (_protocol == "TCP")
-        listen(sockfd, 5);
-      _serverSocket = sockfd;
-      logger_->log_info("ListenSysLog Server socket %d bind OK to port %d", _serverSocket, portno);
-    }
-    FD_ZERO(&_readfds);
-    FD_SET(_serverSocket, &_readfds);
-    _maxFds = _serverSocket;
-    std::vector<int>::iterator it;
-    for (it = _clientSockets.begin(); it != _clientSockets.end(); ++it) {
-      int clientSocket = *it;
-      if (clientSocket >= _maxFds)
-        _maxFds = clientSocket;
-      FD_SET(clientSocket, &_readfds);
-    }
-    fd_set fds;
-    struct timeval tv;
-    int retval;
-    fds = _readfds;
-    tv.tv_sec = 0;
-    // 100 msec
-    tv.tv_usec = 100000;
-    retval = select(_maxFds + 1, &fds, NULL, NULL, &tv);
-    if (retval < 0)
+  int port = Port.getDefaultValue();
+  if (!context->getProperty(Port.getName(), port)) {
+    logger_->log_error("Missing or invalid Port: defaulting to %s", port);
```

Comment:
The same concerns here as in case of the protocol

Reply:
Good point. I checked and since these are required properties the context->getProperty will already throw if it can't find the property.
So these branches are unreachable. I removed them in https://github.com/apache/nifi-minifi-cpp/pull/1294/commits/251865e006f1e53e45c33fdaa184a9259f7bf13d#diff-a22ae2c37869a184bd936ac719f98ae8bbdb1b8d2ad5b3e6fc48ea9207fe530bR95-R99